FHICORC.Backend/FHICORC.Application.Services/RevocationService.cs

```diff
@@ -53,6 +60,140 @@ public RevocationService(ILogger<RevocationService> logger, CoronapassContext co
+                    
+                    CreateBatch(newGuid.ToString());
+                    CreateHash(newGuid.ToString(), hash.value);
+                }
+            }
+        }
+
+        private void CreateHash(string batchId, string hash)
+        {
+            var hashDto = new RevocationHash
+            {
+                BatchId = batchId,
+                Hash = hash,
+            };
+            _coronapassContext.ChangeTracker.Clear();
+            _coronapassContext.RevocationHash.Add(hashDto);
+            _coronapassContext.SaveChanges();
+        }
+        private void CreateBatch(string batchId)
+        {
+            var newBatch = new RevocationBatch
+            {
+                BatchId = batchId,
+                Expires = DateTime.Now.AddMonths(3),
+                Country = COUNTRY_CODE,
+                HashType = HASH_TYPE,
+                Deleted = false,
+                Upload = true,
+            };
+            _coronapassContext.ChangeTracker.Clear();
+            _coronapassContext.RevocationBatch.Add(newBatch);
+            try
+            {
+
+                _coronapassContext.SaveChanges();
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Duplicate Key: {batchId}, message: {ex.Message}");
+            }
+        }
+        private List<string> ReturnUniqueHashes(List<string> hashList, HashesDto revokedHashList)
+        {
+            return hashList
+                .Where(h => revokedHashList.Hashes
+                .All(rh => rh.HashInfo != h))
+                .ToList();
+        }
+        private HashesDto FetchHashes()
+        {
+            var hashList = _coronapassContext.RevocationHash
+                .Select(x => new Hash()
+                {
+                    Id = x.Id,
+                    BatchId = x.BatchId,
+                    HashInfo = x.Hash,
+                }
+                ).ToList();
+
+            return new HashesDto()
+            {
+                Hashes = hashList
+            };
+        }
+        private BatchItem FetchSmallestBatchItem()
+        {
+            return _coronapassContext.RevocationBatch.ToList()
+                .Join(_coronapassContext.RevocationHash,
+                    b => b.BatchId,
+                    h => h.BatchId, (b, h) => new { Batch = b, Hash = h })
+                .Where(x => x.Batch.Country.Equals(COUNTRY_CODE) && x.Batch.Expires.Date == ExpiryDate)
+                .GroupBy(y => new
+                {
+                    y.Batch.BatchId,
+                    y.Batch.Expires,
+                })
+                .Where(x => x.Count() < MAX_SIZE_BATCH)
+                .Select(y => new BatchItem
+                {
+                    BatchId = y.Key.BatchId,
+                    Count = y.Count(),
+                    Expires = y.Key.Expires,
+                })
+                .OrderBy(x => x.Count)
+                .FirstOrDefault();
+        }
+        class BatchItem
```

Comment:
If this is only used in the class, make private, else move to own file, and determine if it should be internal or public